website/blog/how-to-migrate-an-ios-app-to-bazel.md

```diff
@@ -0,0 +1,1470 @@
+    without having to migrate those to Bazel manually.
+
+Here is the new `Package.swift` file that includes the dependencies that were referenced in the Xcode project:
+
+```swift
+// swift-tools-version:5.7
+
+import PackageDescription
+
+let package = Package(
+    name: "Mastodon-iOS",
+    defaultLocalization: "en",
+    platforms: [
+        .iOS(.v16),
+    ],
+    dependencies: [
+        .package(name: "ArkanaKeys", path: "Dependencies/ArkanaKeys"),
+        .package(name: "MastodonSDK", path: "MastodonSDK"),
+        .package(
+            url: "https://github.com/Bearologics/LightChart.git",
+            branch: "master"
+        ),
+        .package(
+            url: "https://github.com/jdg/MBProgressHUD.git",
+            from: "1.2.0"
+        ),
+        .package(
+            url: "https://github.com/tid-kijyun/Kanna.git",
+            from: "5.2.7"
+        ),
+    ]
+)
+```
+
+Currently **rules_swift_package_manager** requires you to use [Gazelle][gazelle] in order to create a `swift_deps_index.json` file.[^rspm-no-gazelle]
+So we need to add a `bazel_dep` on **gazelle** in addition to **rules_swift_package_manager** in our `MODULE.bazel` file.
+We also need to add a [`use_repo`][use_repo] stanza for the Swift packages we directly depend on
+(though if you run the `//:swift_update_pkgs` Gazelle target,
+which we will define shortly,
+it will add the required stanza for you):
+
+[^rspm-no-gazelle]:
+    There is [a plan][remove-swift_deps_index] that before **rules_swift_package_manager** reaches 1.0.0 it will remove the need for the `swift_deps_index.json` file,
+    and the need to use Gazelle with it.
+
+```python
+bazel_dep(name = "rules_swift_package_manager", version = "0.28.0")
+bazel_dep(name = "gazelle", version = "0.35.0")
+
+# swift_deps START
+swift_deps = use_extension(
+    "@rules_swift_package_manager//:extensions.bzl",
+    "swift_deps",
+)
+swift_deps.from_file(
+    deps_index = "//:swift_deps_index.json",
+)
+use_repo(
+    swift_deps,
+    "swiftpkg_arkanakeys",
+    "swiftpkg_arkanakeysinterfaces",
+    "swiftpkg_kanna",
+    "swiftpkg_lightchart",
+    "swiftpkg_mastodonsdk",
+    "swiftpkg_mbprogresshud",
+)
+# swift_deps END
+```
+
+And now we can define our Gazelle targets in a root `BUILD` file:
+
+```python
+load("@gazelle//:def.bzl", "gazelle", "gazelle_binary")
+load(
+  "@rules_swift_package_manager//swiftpkg:defs.bzl",
+  "swift_update_packages",
+)
+
+# - Gazelle
+
+# Ignore the `.build` folder that is created by running Swift package manager
+# commands. The Swift Gazelle plugin executes some Swift package manager
+# commands to resolve external dependencies. This results in a `.build` file
+# being created.
+# NOTE: Swift package manager is not used to build any of the external packages.
+# The `.build` directory should be ignored. Be sure to configure your source
+# control to ignore it (i.e., add it to your `.gitignore`).
```

Comment:
Should this be added to the `.gitignore` on the branch https://github.com/mastodon/mastodon-ios/compare/develop...brentleyjones:mastodon-ios:bj/migrate-to-mastodon#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947?